### ISSUE TITLE
Append new config files to existing configs

### ISSUE THREAD

**Issue by seanchen1991** [informalsystems/hermes#3520]
## Summary

From a discussion with Lavender 5 about Hermes' `config auto` capabilities, it would be a nice quality-of-life feature if newly-generated config files could be automatically appended to an existing config file. This would avoid users outputting a new config file with the `config auto` command and then have to manually copy/paste the content of the generated config onto their pre-existing config file. 

## Proposal

The `config auto` CLI could perhaps be changed to accept an `--append` flag that would take a path to a pre-existing config file as a parameter. The contents of the newly-generated config would then be appended onto this pre-existing config file.  

## Acceptance Criteria

- [ ] The `config auto` command exhibits the desired append behavior

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by seanchen1991** [comment]
Yea, that make sense. 

> support stdin as output

Can you expand a bit on what you mean by this?

**Comment by romac** [comment]
Sorry I meant stdout...

I just gave it a try with

```
$ hermes config auto --output /dev/stdout --chains agoric --chains osmosis | tee test.toml
```

and the result `test.toml` file looks good except for the last line, which reads:

```
SUCCESS Config file written successfully at '/dev/stdout'
```

For most commands, it is by design that only the last output line, which indicates whether or not the command succeeded, is output to stdin, while all other log messages are output to stderr.

We could either:

1. Change this behavior for the `config auto` command and output all log messages to stderr
2. Add special support for stdout to the command, eg. by special-casing the `/dev/stdout` or `-` value for the `--output` parameter and only emit the last message on stderr in that case.
3. Change this behavior for all commands and output all log messages to stderr

(3) would be a breaking change and might prevent some operators from using Hermes in scripts, or break their setups, etc. so it's imho a no go.

Either (1) or (2) sound good to me, with a preference for (1) as it's simpler to implement and we don't expect operators to use complex scripts to invoke the `config auto` command as they probably have to review the config manually anyway.

What do you think?

**Comment by seanchen1991** [comment]
Option 1 sounds good me 🙂

**Comment by seanchen1991** [comment]
The way I understood the request was that a user should be able to append the output of a `config auto` call onto an existing `config.toml` file. I was imagining a flow like this:
1. Generate a fresh config file with `hermes config auto --output test.toml --chains agoric`
2. Generate the config for a different chain and append the output onto `test.toml` with `hermes config auto --output /dev/stdout --chains osmosis >> test.toml`
Doing it this way creates a bunch of duplicated lines, as both `config auto` calls generate a bunch of config sections, such as `[global]`, `[telemetry]`, etc. Ideally, with this flow, only the `[[chains]]` section from the second `config auto` call would get appended to the pre-existing config file. I attempted to deduplicate the file by piping it to `uniq -u`, but that didn't see to do anything. 

So with all that said, I'm not sure your suggested options will achieve the flow we're looking for, at least as far as how I understand it.

**Comment by romac** [comment]
Good point, I had somehow missed that. Then yeah having bespoke flag for only outputting the chain config sounds like the way to go!
